src/perlobj.jl

```diff
@@ -19,16 +19,20 @@ function bigobj(name::String, input_data::Pair{<:Union{Symbol,String}}...; kwarg
     return obj
 end
 
-Base.propertynames(p::Polymake.BigObject) = Symbol.(Polymake.complete_property(p, ""))
-
-function Base.setproperty!(obj::BigObject, prop::String, val)
-    return take(obj, prop, convert(PolymakeType, val))
-end
+Base.propertynames(p::Polymake.BigObject) = Symbol.(complete_property(p, ""))
 
 function Base.setproperty!(obj::BigObject, prop::Symbol, val)
+    @assert prop != :cpp_object
     return take(obj, string(prop), convert(PolymakeType, val))
 end
 
+function Base.setproperty!(obj::BigObject, prop::Symbol, val::Ptr{Nothing})
```

Comment:
Isn't `Ptr{Cvoid}` more idiomatic (in the C sense) than `Ptr{Nothing}` (I know they are the same on the julia side but still...)
```suggestion
function Base.setproperty!(obj::BigObject, prop::Symbol, val::Ptr{Cvoid})
```